overlord/snapstate/backend/aliases.go

```diff
@@ -46,12 +46,13 @@ func (b Backend) UpdateAliases(add []*Alias, remove []*Alias) error {
 			return fmt.Errorf("cannot remove alias symlink: %v", err)
 		}
 		removed[alias.Name] = true
-		completer := filepath.Join(dirs.CompletersDir, alias.Name)
-		target, err := os.Readlink(completer)
-		if err != nil || target != alias.Target {
-			continue
+		for _, completersPath := range []string{dirs.CompletersDir, dirs.LegacyCompletersDir} {
+			completer := filepath.Join(completersPath, alias.Name)
+			target, err := os.Readlink(completer)
+			if err == nil || target == alias.Target {
```

Comment:
If you change the logic here to remove inside the `if` you need a also to change the `||` to `&&`. Anyway, I think that the coding conventions in snapd prefer in general to use `continue` when some condition inside the loop is not met.

Reply:
Oops. Thank you for spotting that.